tensorflow_addons/metrics/r_square.py

```diff
@@ -139,7 +139,8 @@ def result(self) -> tf.Tensor:
 
     def reset_states(self) -> None:
         # The state of the metric will be reset at the start of each epoch.
-        self.squared_sum.assign(0)
-        self.sum.assign(0)
-        self.res.assign(0)
-        self.count.assign(0)
+        reset_value = 0
+        self.squared_sum.assign(reset_value)
+        self.sum.assign(reset_value)
+        self.res.assign(reset_value)
+        self.count.assign(reset_value)
```

Comment:
using `0` directly will not have duplicate initial value generation call :-)

Reply:
Yes but it could simplify semantic readibility if all the values need to be the same